src/Engine/ProtoCore/Core.cs

```diff
@@ -1510,17 +1510,20 @@ public void Cleanup()
 
         public void InitializeContextGlobals(Dictionary<string, object> context)
```

Comment:
Is this used anywhere? Consider killing it for simplification if it isn't. Given that it's limited to doubles, I'm very sceptical.